src/subdomains/generic/kyc/services/kyc.service.ts

```diff
@@ -306,6 +349,7 @@ export class KycService {
 
     let user = transaction.user;
     const kycStep = user.getStepOrThrow(transaction.stepId);
+    const nationalityMatch = dto.userdata.nationality.value !== user.nationality?.symbol;
 
```

Comment:
Do the nationality check in `getIdentCheckErrors` -> add a new `IdentCheckError` (step will go to manual review)